Packs/MicrosoftGraphSearch/Integrations/MicrosoftGraphSearch/MicrosoftGraphSearch_description.md

```diff
@@ -0,0 +1,8 @@
+# Authentication
+### Authentication Based on Azure Active Directory Applications
+
+Microsoft integrations (Graph and Azure) in Cortex XSOAR use Azure Active Directory applications to authenticate with Microsoft APIs. These integrations use OAuth 2.0 and OpenID Connect standard compliant authentication services, which use an application to sign in or delegate authentication. For more information, see the Microsoft identity platform overview.
+
+There is only one application authentication method available:
+
+ * [Self-Deployed Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#self-deployed-application)
```

Comment:
The application authentication you implemented is a self-deployed one, but the authentication method is the Authorization code flow - see [this](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#authorize-on-behalf-of-a-user).
If this is the flow you want to implement, I would recommend adding the '<>-login-url' command so it would be easier to use.
You can also try and adding the Client Credentials authentication flow - which is the easiest to configure (but again, just a recommendation). If you decide to do one of those - you can contact me here or on slack for more details and help :)

Reply:
I'm willing to try using Client Credentials flow if it works for this API

Reply:
@randomizerxd I think it would b the easier solution, too.

Reply:
I believe this API requires delegated permissions as described in this doc: https://learn.microsoft.com/en-us/graph/api/resources/search-api-overview?view=graph-rest-1.0

Reply:
@randomizerxdSo that's why it didn't work for us with the Client Credentials 😅